src/main/java/com/massivecraft/factions/zcore/persist/MemoryBoard.java

```diff
@@ -165,12 +165,11 @@ public void clean(String factionId) {
                 Entry<FLocation, String> entry = iter.next();
                 if (entry.getValue().equals(factionId)) {
                     LWCFeatures.clearAllChests(entry.getKey());
-                    iter.remove();
                 }
             }
-        } else {
-            flocationIds.removeFaction(factionId);
         }
+
+        flocationIds.removeFaction(factionId);
```

Comment:
@CmdrKittens I believe this is the correct way to do this. Thoughts?

Reply:
Yeah, this should work out. 👍